.github/workflows/ci.yml

```diff
@@ -140,13 +140,22 @@ jobs:
           cache: "pip"
 
       - name: Installing Linux Dependencies
-        if: ${{ runner.os == 'Linux' && matrix.os != 'ubuntu-20.04' && !env['TEST_X86'] }}
+        if: ${{ runner.os == 'Linux' && matrix.os != 'ubuntu-22.04' && !env['TEST_X86'] }}
         run: |
           sudo apt update
-          sudo apt install cmake clang-14 clang-tools llvm kcov g++-12 valgrind zlib1g-dev libcurl4-openssl-dev
+          sudo apt install cmake clang-15 clang-tools llvm g++-12 valgrind zlib1g-dev libcurl4-openssl-dev
+          # Install kcov from source
+          sudo apt-get install binutils-dev libssl-dev libelf-dev libstdc++-12-dev libdw-dev libiberty-dev
+          git clone https://github.com/SimonKagstrom/kcov.git
+          cd kcov
+          mkdir build
+          cd build
+          cmake ..
+          make
+          sudo make install
```

Comment:
kcov is not available in 24.04 (yet)
see https://answers.launchpad.net/ubuntu/+source/kcov/+question/818841

Reply:
I'm not sure if this manual installation is the way to go here, seems a bit out-of-place but might just be a necessary evil. Alternatively we could wait to update the `22.04` runners to `24.04` and just keep them as-is.

Reply:
There is no need to have the `kcov` build run on `24.04`. This is primarily a coverage run, which should produce the same results on both distro versions. On the other hand, it looks like `24.04` will never get that package, and if building `kcov` is relatively trivial and works, then why not go with that build?

Reply:
btw, if you think that the build script for `kcov` is unnecessarily detailed for the workflow spec, you can move it to the scripts folder and only invoke the script from the workflow.

Reply:
alright, then I'll keep it as it is now (part of the `24.04` run, and since the `kcov` install is pretty minimal it can stay in the workflow file)